.github/workflows/ci.yml

```diff
@@ -0,0 +1,129 @@
+# This workflow will build a .NET project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net
+
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+  workflow_dispatch:
+
+permissions:
+  pull-requests: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  DOTNET_CLI_PERF_LOG: false
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+  DOTNET_GENERATE_ASPNET_CERTIFICATE: false
+  DOTNET_NOLOGO: true
+  SuppressNETCoreSdkPreviewMessage: true
+  SLN_PATH: source/FlashOWare.Generators.sln
+
+jobs:
+  build:
+    name: Restore & Build & Test
+    continue-on-error: false
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        configuration: [Debug, Release]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        global-json-file: global.json
+    - name: Restore dependencies
+      run: dotnet restore ${{ env.SLN_PATH }} --configfile nuget.config --no-cache
+    - name: Build
+      run: dotnet build ${{ env.SLN_PATH }} --configuration ${{ matrix.configuration }} --no-restore
+    - name: Test
+      run: dotnet test ${{ env.SLN_PATH }} --configuration ${{ matrix.configuration }} --no-build
+  codeCoverage:
+    name: Collect Code Coverage
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    env:
+      COVERAGE_OUTPUT_PATH: artifacts/coverage
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          global-json-file: global.json
+      - name: Restore dependencies
+        run: dotnet restore ${{ env.SLN_PATH }} --configfile nuget.config --no-cache
+      - name: Build
+        run: dotnet build ${{ env.SLN_PATH }} --configuration Release --no-restore
+      - name: Test
+        run: dotnet test ${{ env.SLN_PATH }} --configuration Release --no-build --collect:"XPlat Code Coverage" --logger trx --results-directory ${{ env.COVERAGE_OUTPUT_PATH }}
```

Comment:
Since we already run the "holy trinity" in the `Restore & Build & Test` job, which is also a dependency for this code-coverage job, I'd be fine to "merge" them here into just a `dotnet test` for brevity.
Or is there a (potential) disadvantage?

Reply:
I changed my mind again ... I do prefer the explicit steps here as well now, so that we have more control over them.